slashing-protection/referencetests/src/test/java/tech/pegasys/web3signer/slashingprotection/ReferenceTestRunner.java

```diff
@@ -0,0 +1,155 @@
+import org.junit.jupiter.api.Test;
+
+public class ReferenceTestRunner {
+
+  private static final Logger LOG = LogManager.getLogger();
+
+  private static final ObjectMapper objectMapper =
+      new ObjectMapper()
+          .registerModule(new InterchangeModule())
+          .enable(MapperFeature.ACCEPT_CASE_INSENSITIVE_ENUMS);
+  private final ValidatorsDao validators = new ValidatorsDao();
+  private EmbeddedPostgres slashingDatabase;
+
+  @Before
+  void setup() throws IOException {
+    slashingDatabase = EmbeddedPostgres.start();
+    final Flyway flyway =
+        Flyway.configure()
+            .locations("/migrations/postgresql/")
+            .dataSource(slashingDatabase.getPostgresDatabase())
+            .load();
+    flyway.migrate();
+  }
+
+  @After
+  void cleanup() {
+    try {
+      slashingDatabase.close();
+    } catch (final IOException e) {
+      LOG.error("Failed to close database", e);
+    }
+  }
+
+  @Test
+  @Disabled("Until interchange import is available")
+  void execute() throws IOException {
+    final URL refTestPath =
+        Resources.getResource(
+            Path.of("slashing-protection-interchange-tests", "tests", "generated").toString());
+
+    final Path testFilesPath = Path.of(refTestPath.getPath());
+    try (Stream<Path> testFiles = Files.list(testFilesPath)) {
+      testFiles.forEach(this::executeFile);
+    }
+  }
+
+  private void executeFile(final Path inputFile) {
+    LOG.info(inputFile.toString());
+    try {
+      final TestFileModel model = objectMapper.readValue(inputFile.toFile(), TestFileModel.class);
+      final String interchangeContent =
+          objectMapper.writeValueAsString(model.getInterchangeContent());
+
+      setup();
+      final String databaseUrl =
+          String.format("jdbc:postgresql://localhost:%d/postgres", slashingDatabase.getPort());
+
+      final SlashingProtection slashingProtection =
+          SlashingProtectionFactory.createSlashingProtection(databaseUrl, "postgres", "postgres");
+
+      slashingProtection.importData(new ByteArrayInputStream(interchangeContent.getBytes(UTF_8)));
+
+      final List<String> validatorsInImport =
+          model.getInterchangeContent().getSignedArtifacts().stream()
+              .map(SignedArtifacts::getPublicKey)
+              .collect(Collectors.toList());
+
+      final Jdbi jdbi = DbConnection.createConnection(databaseUrl, "postgres", "postgres");
+      final List<String> publicKeysInDb =
+          jdbi.withHandle(
+              h ->
+                  validators
+                      .findAllValidators(h)
+                      .map(v -> v.getPublicKey().toHexString())
+                      .collect(Collectors.toList()));
+
+      assertThat(validatorsInImport).containsExactlyInAnyOrderElementsOf(publicKeysInDb);
+
+      // need to register the validators with slashingProtection before testing blocks/attestations
+      if (validatorsInImport.isEmpty()) {
+        return;
```

Comment:
Why exit early here? Is this an error if there are no validators? I remember Web3Signer doesn't start if there are no validators but this seems the wrong to fix that.

Reply:
if there are no validators, then we can't register them - i.e.slashingprotection normally registers a list of validators automagically - if its zero, things go bad - so no validators, no tests.